core/agent.py

```diff
@@ -63,19 +63,24 @@ def get_services_status(self, dialog_id: str):
         return done, waiting
 
     def process_service_response(self, dialog_id: str, service_name: str = None, response: Any = None,
-                                 response_time: float = None):
+                                 **kwargs):
         workflow_record = self.get_workflow_record(dialog_id)
 
         # Updating workflow with service response
         service = self.pipeline.get_service_by_name(service_name)
         if service:
-            service_data = self.workflow[dialog_id]['services'][service_name]
+            service_data = workflow_record['services'][service_name]
             service_data['done'] = True
-            service_data['done_time'] = response_time
+            service_data['done_time'] = time()
```

Comment:
I think it's better to rename this one to a more obvious name, for example, `agent_done_time` or `agent_received_time`. Otherwise, it might be confused with `service_response_time`